### DISCUSSION TITLE
D3 library and worklets.

### DISCUSSION THREAD

**Discussion by david-laurentino** [software-mansion/react-native-reanimated#1600] [Q&A]
Is it possible to use libraries like d3 inside worklets or hooks? I've been trying things like:

```
function arcWorket(startAngle, endAngle) {
         'worklet';
         const arc = d3.arc()
        .innerRadius(210)
        .outerRadius(310)
        .startAngle(([startAngle, endAngle]) => startAngle)
        .endAngle(([startAngle, endAngle]) => endAngle)
        return arc([startAngle, endAngle]);
}
```

```
const animatedProps = useAnimatedProps(() => {
        const arc = d3.arc()
              .innerRadius(210)
              .outerRadius(310)
              .startAngle(([startAngle, endAngle]) => startAngle)
              .endAngle(([startAngle, endAngle]) => endAngle)
        const path = arc([startAngle, sharedProgress.value * endAngle]);

        return {
            d: path,
        };
});
```

And then the app crashes displaying error messages from native code. Worklets and hooks must not call any function from external libraries?

**Comment by karol-bisztyga** [comment]
> And then the app crashes displaying error messages from native code

What error messages exactly?

**Reply by david-laurentino** [reply]
![Screenshot_2021-01-07-13-57-59-879_com miui bugreport](https://user-images.githubusercontent.com/18560166/103921275-11265f80-50f1-11eb-8a25-6060737bbafc.jpg)

The full stack is very large, but I can send it if you want.
